src/core/api/clock.ts

```diff
@@ -345,18 +352,30 @@ export interface IClockOptions {
 function createClock(
   mediaElement : HTMLMediaElement,
   options : IClockOptions
-) : Observable<IClockTick> {
-  return observableDefer(() : Observable<IClockTick> => {
+) : { clock$: Observable<IClockTick>, setCurrentTime: (time: number) => void } {
+  let internalSeek: number | null = null;
+  const setCurrentTime = (time: number) => {
+    internalSeek = time;
```

Comment:
We set the time but never uses it (we check it as if it was a boolean).

---

After thinking a little more about it, I'm unsure of whether we should set it to a `boolean` or to a `number.

On one hand, setting it to a `number` will allows to still have the "SEEKING" state (and "seeking" player event) when another synchronous seek was done to another one position but only one "seeking" event was sent by the browser for some reasons (we are not even sure if this can happen)  but on the other hand, I guess that when doing two synchronous seeks the `currentTime` will be set to the last one even when the first event is received, so we should not be really able to use it for what we want.

What do you think?